webapp/src/components/modals/create_issue/create_issue.jsx

```diff
@@ -0,0 +1,205 @@
+                issueTitleValid: Boolean(this.state.issueTitle),
+                showErrors: true,
+            });
+            return;
+        }
+
+        const issue = {
+            title: this.state.issueTitle,
+            body: this.props.post.message,
+            repo: this.state.repoValue,
+            post_id: this.props.post.id,
+        };
+
+        this.setState({submitting: true});
+
+        this.props.create(issue).then((created) => {
+            if (created.error) {
+                this.setState({error: created.error.response.body.message, submitting: false});
+                return;
+            }
+            this.handleClose(e);
+        });
+    };
+
+    handleClose = (e) => {
+        if (e && e.preventDefault) {
+            e.preventDefault();
+        }
+        this.setState(initialState, this.props.close);
+    };
+
+    handleRepoValueChange = (name) => {
+        this.setState({
+            repoValue: name,
+        });
+    };
+
+    handleIssueTitleChange = (newValue) => {
+        this.setState({
+            issueTitle: newValue,
+        });
+    };
+
+    render() {
+        const {visible, theme} = this.props;
+        const {error, submitting} = this.state;
+        const style = getStyle(theme);
+
+        const requiredMsg = 'This field is required.';
+        let issueTitleValidationError = null;
+        if (this.state.showErrors && !this.state.issueTitleValid) {
+            issueTitleValidationError = (
+                <p className='help-text error-text'>
+                    <span>{requiredMsg}</span>
+                </p>
+            );
+        }
+
+        if (!visible) {
+            return null;
+        }
+
+        let submitError = null;
+        if (error) {
+            submitError = (
+                <p className='help-text error-text'>
+                    <span>{error}</span>
+                </p>
+            );
+        }
+
+        const component = (
+            <div>
+                <GithubRepoSelector
+                    onChange={this.handleRepoValueChange}
+                    value={this.state.repoValue}
+                    required={true}
+                    theme={theme}
+                    addValidate={this.validator.addComponent}
+                    removeValidate={this.validator.removeComponent}
+                />
+                <Input
+                    id={'title'}
+                    label='Title for the GitHub Issue'
+                    type='input'
+                    required={true}
+                    disabled={false}
+                    maxLength={65}
+                    value={this.state.issueTitle}
+                    onChange={this.handleIssueTitleChange}
+                />
+                {issueTitleValidationError}
+                <Input
+                    label='Description for the GitHub Issue'
+                    type='textarea'
+                    value={this.props.post.message}
+                    disabled={true}
```

Comment:
Why can't the user edit the description?

Reply:
Done